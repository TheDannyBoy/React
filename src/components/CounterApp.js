import PropTypes from 'prop-types';
import { useState } from 'react';


const CounterApp = ({ initialValue }) => {

  const [counter, setCounter] = useState(0);

  const handleReset = () => setCounter(initialValue);
  const handleClick = () => setCounter( count => count + 1);
  const handleSubstract = () => setCounter( count => count - 1);

  return <>
    <h1>CounterApp</h1>
    <h2> { counter } </h2>
    <button onClick={handleSubstract}>-1</button>
    <button onClick={handleReset}>Reset</button>
    <button onClick={handleClick}>+1</button>
  </>
}


CounterApp.propTypes = {
  initialValue: PropTypes.number,
}

CounterApp.defaultProps = {
  initialValue: 0,
}

export default CounterApp;
