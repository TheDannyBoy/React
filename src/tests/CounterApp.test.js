import { shallow } from 'enzyme'
import CounterApp from '../components/CounterApp'

describe('Pruebas en CounterApp', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test('CounterApp debe crearse correctamente y mostrar el contador por defecto', () => {
    const valorContador = wrapper.find('h2').text().trim();

    expect(wrapper).toMatchSnapshot();
    expect(valorContador).toBe('10');
  });

  test('Al pasar el valor del contador como prop, debería visualizarse ese mismo valor', () => {
    const wrapper = shallow(<CounterApp initialValue={100}/>)
    const valorContador = wrapper.find('h2').text().trim();

    expect(wrapper).toMatchSnapshot();
    expect(valorContador).toBe('100');
  });

  test('Debería incrementar en 1 el contador al darle click al botón +1', () => {
    wrapper.find('button').at(2).simulate('click');
    const valorContador = wrapper.find('h2').text().trim();
    expect(valorContador).toBe('11');
  });

  test('Debería decrementar en 1 el contador al darle click al botón -1', () => {
    wrapper.find('button').at(0).simulate('click');
    const valorContador = wrapper.find('h2').text().trim();
    expect(valorContador).toBe('9');
  });

  test('Debería resetear el contador al darle click al reset', () => {
    const wrapper = shallow(<CounterApp initialValue={1000}/>)
    wrapper.find('button').at(2).simulate('click');
    wrapper.find('button').at(1).simulate('click');
    const valorContador = wrapper.find('h2').text().trim();
    expect(valorContador).toBe('1000');
  });


});
