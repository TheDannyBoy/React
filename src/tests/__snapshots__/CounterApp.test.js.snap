// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp Al pasar el valor del contador como prop, deber√≠a visualizarse ese mismo valor 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;

exports[`Pruebas en CounterApp CounterApp debe crearse correctamente y mostrar el contador por defecto 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;
